api/samples/neoncounter.c

```diff
@@ -0,0 +1,893 @@
+/* ******************************************************************************
+ * Copyright (c) 2014-2018 Google, Inc.  All rights reserved.
+ * Copyright (c) 2011 Massachusetts Institute of Technology  All rights
+ * reserved. Copyright (c) 2008 VMware, Inc.  All rights reserved.
+ * ******************************************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of VMware, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/*
+ * Counter for NEON/SIMD-instructions on AARCH64, based on the inscount.cpp
+ * sample
+ * University of Regensburg, Germany (QPACE 4, SFB TRR-55)
+ */
+
+#include "dr_api.h"
+#include "dr_tools.h"
+#include "drmgr.h"
+
+#include "string.h"
+#include <stdio.h>
+
+#ifdef DEBUG
+#    define IS_DEBUG true
+#else
+#    define IS_DEBUG false
+#endif
+
+// in DEBUG mode all processed instructions are written in this file
+FILE *debug;
+
+// remembers if only the app or the app and all shared libraries should be
+// analyzed
+static bool app_only = false;
+// application module
+static app_pc exe_start;
+
+// saves the name of the executed app
+static const char *executable;
+
+// saves the name of the used client library
+const char *library;
+
+// ARM-Architecture Reference Book p. 185ff
+// Arithmetic instructions:
+int OP_arithmetic[] = {
```

Comment:
The list of AArch64 opcodes can be found in build directory: `build/include/dr_ir_opcodes_aarch64.h`